### PR TITLE
refine aiter_backend for mtp

### PR DESCRIPTION
**Pull request by valarLip**
1. refine aiter_backend, for mtp
2. enable aiter_biased_grouped_topk kernel (addressed other place)
3. take aiter get_rope back (addressed other place)
4. enable aiter fp8 block scale quant (addressed other place)

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Comment by valarLip** [comment]
server launch:
`SGLANG_USE_AITER=1 \
python3 -m sglang.launch_server \
--model-path deepseek-ai/DeepSeek-R1-0528 \
--attention-backend aiter \
--port 30000 --host 0.0.0.0 --trust-remote-code --tp-size 8 --enable-metrics --mem-fraction-static 0.85 --chunked-prefill-size 131072 \
--speculative-algorithm NEXTN --speculative-draft-model-path SGLang/DeepSeek-R1-NextN --speculative-num-steps 2 --speculative-eagle-topk 1 --speculative-num-draft-tokens 3 --speculative-accept-threshold-single=0.001`

accuracy test:
`python3 sglang/benchmark/gsm8k/bench_sglang.py --num-questions 200 --parallel 50
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 200/200 [00:23<00:00,  8.36it/s]
Accuracy: 0.965
Invalid: 0.000
Latency: 24.789 s
Output throughput: 751.491 token/s`

**Comment by HaiShaw** [comment]
@valarLip please cross-check https://github.com/sgl-project/sglang/pull/7240 for the 2nd enhancement.
cc @kkHuang-amd

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 1]
```diff
@@ -42,7 +42,10 @@
 _use_aiter = get_bool_env_var("SGLANG_USE_AITER") and _is_hip
 
 if _use_aiter:
-    from aiter import gemm_a8w8_blockscale_CK
+    import aiter
+    from aiter import gemm_a8w8_blockscale_CK, get_hip_quant
+
+    aiter_per1x128_quant = get_hip_quant(aiter.QuantType.per_1x128)
```
Comment (kkHuang-amd) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: Below code does not use it, could we remove it

**Comment by valarLip** [comment]
> @valarLip please cross-check #7240 for the 2nd enhancement. cc @kkHuang-amd

this way is my old implementation... we can switch to this new one now

File: python/sglang/srt/layers/quantization/fp8_utils.py (hunk 1 above)
Comment (valarLip) [reply python/sglang/srt/layers/quantization/fp8_utils.py]: will use it later, just leave a placeholder for now

File: python/sglang/srt/layers/attention/aiter_backend.py [hunk 2]
```diff
@@ -157,13 +162,13 @@ def init_forward_metadata(self, forward_batch: ForwardBatch):
...
         kv_last_page_len = None
-        max_extend_len = None
+        max_q_len = None
 
         if forward_batch.forward_mode.is_decode_or_idle():
             if spec_info is None:
                 kv_indptr[1 : bs + 1] = torch.cumsum(forward_batch.seq_lens, dim=0)
                 kv_indptr = kv_indptr[: bs + 1]
-                kv_indices = torch.zeros(
+                kv_indices = torch.empty(
```
Comment (HaiShaw) [inline python/sglang/srt/layers/attention/aiter_backend.py:171]: ROCm: torch.empty seems to be more troublesome than torch.zeros

**Review by HaiShaw — CHANGES_REQUESTED** [review]
@valarLip
Pls split MTP to another PR, and we plan to have `aiter_biased_grouped_topk` merged first - with that we expect the numbers ~1.96 req/s as https://github.com/sgl-project/sglang/pull/7240

**Comment by valarLip** [comment]
> @valarLip Pls split MTP to another PR, and we plan to have `aiter_biased_grouped_topk` merged first - with that we expect the numbers ~1.96 req/s as #7240

added in this one https://github.com/sgl-project/sglang/pull/7423

File: python/sglang/srt/layers/attention/aiter_backend.py (hunk 2 above)
Comment (valarLip) [reply python/sglang/srt/layers/attention/aiter_backend.py:171]: it's ok for this case, we fill the whole tensor in next few lines. using zeros we will have a addition fill kernel launch...

**Comment by HaiShaw** [comment]
@valarLip with 2., 3., 4. addressed, please repurpose this PR just for 1. - MTP.
cc @kkHuang-amd

**Review by HaiShaw — APPROVED** [review]
LGTM

**Comment by HaiShaw** [comment]
@saienduri - One failed CI

**Comment by saienduri** [comment]
> @saienduri - One failed CI

Unrelated, seeing same failure on main: https://github.com/sgl-project/sglang/actions/runs/16818425796/job/47640357056#step:4:39.

**Review by merrymercy — APPROVED** [review]
